mlx/io/load.h

```diff
@@ -36,77 +38,102 @@ class Writer {
 
 class FileReader : public Reader {
  public:
-  explicit FileReader(std::ifstream is)
-      : is_(std::move(is)), label_("stream") {}
   explicit FileReader(std::string file_path)
-      : is_(std::ifstream(file_path, std::ios::binary)),
-        label_(std::move(file_path)) {}
+      : fd_(open(file_path.c_str(), O_RDONLY)), label_(std::move(file_path)) {}
+
+  ~FileReader() {
+    close(fd_);
+  }
 
   bool is_open() const override {
-    return is_.is_open();
+    return fd_ > 0;
```

Comment:
I think it is fine to keep this as is instead of checking with `fcntl(fd_, F_GETFD)`. We are managing the whole lifecycle of the file descriptor so we will fail on `read` instead.

Perhaps it had some point when a stream could be passed in the constructor so we wouldn't know if it was open/closed/good etc.